### PR TITLE
Add download button for output images

### LINKED ISSUE #3155: Add a download button for images
- [x] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
Adapted from this slack [thread](https://huggingface.slack.com/archives/C02SPHC1KD1/p1675875316318549)

> Should we add a download button for images?

>Reasons not to do it:
> * You can pipe the image to a separate gr.File component and that will have a download button
> * You can right click on the image and click on Save Image As
>
> Reasons to do it:
> * Better user experience. Some users may not know about Save Image As
> * The UI could look cleaner if there isn't a separate component just for downloading?

We are in favor of adding it.

**Describe the solution you'd like**  
Comments from @pngwn 

> I think we should add it by default when it is an 'output' (not interactive) but it could be toggleable via a kwarg when it is in interactive mode (edited) 

> the UI for it can be a download icon (that little arrow thing) and appear where the delete/ clear icons appear in static mode and alongside them in interactive mode

**Additional context**

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Fixes #3155

You can test with `blocks_mask` demo.

download_sketch

After discussing with @pngwn , we decided to only apply this to output images as there are some ambiguities with sketch inputs (namely whether the mask or image should be downloaded). This feature has come up in the context of output images, so we think this covers most cases and we can add support for more cases if there is user demand and the image component gets refactored a bit.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: ui/packages/image/src/StaticImage.svelte [hunk 1]
```diff
@@ -19,6 +20,15 @@
 {#if value === null}
 	<Empty size="large" unpadded_box={true}><Image /></Empty>
 {:else}
+	<div class="download">
+		<a
+			href={value}
+			target={window.__is_colab__ ? "_blank" : null}
+			download={"image"}
```
Comment (freddyaboulton) [inline ui/packages/image/src/StaticImage.svelte:27]: We don't have the filename generated on the python side so I'm using `image` as the filename. I'm not sure we'd even want the tempfile name here anyways.

**Review by pngwn — APPROVED** [review]
lgtm apart from the changelog weirdness!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,15 @@
 # Upcoming Release
 
 ## New Features:
+
+### Download button for images 🖼️
+
+Output images will now automatically have a download button displayed to make it easier to save and share
+the results of Machine Learning art models.
```
[the remark this answers is not in the document]
Comment (pngwn) [reply CHANGELOG.md:8]: or is it a haiku

File: ui/packages/image/src/StaticImage.svelte (hunk 1 above)
Comment (abidlabs) [reply ui/packages/image/src/StaticImage.svelte:27]: Can we use the stem from `{value}` here as the filename? I'll be working on #943 probably next week which will make temporary file names be the same as the original filenames (in those cases where we have the original filenames) so I think it's better if we use the temp names.

**Comment by abidlabs** [comment]
Very handy! I did have a suggestion above, but besides that LGTM!

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (freddyaboulton) [reply CHANGELOG.md:8]: I'm a mysterious poet 🖊️ 

I just broke it up so it'd be easier to read in a text editor. There isn't a new line in the rendered version I think.

image

File: ui/packages/image/src/StaticImage.svelte (hunk 1 above)
Comment (abidlabs) [reply ui/packages/image/src/StaticImage.svelte:27]: Hold on, take that back. `{value}` is base64 isn't it? I think we don't create temporary files for images like we do for the other components

Comment (abidlabs) [reply ui/packages/image/src/StaticImage.svelte:27]: So nvm, this PR LGTM

**Comment by freddyaboulton** [comment]
Thank you for the reviews @pngwn and @abidlabs !

**Comment by naotokui** [comment]
Hi, is it possible to hide the download button?

**Comment by ruleGreen** [comment]
> Hi, is it possible to hide the download button?

I also want to hide the download button

**Pull request merged**